### PR TITLE
[GEOT-7102] ElasticSearch plugin move aggregation creation from view parameters to Rendering Transformation parameter

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOT-7102](https://osgeo-org.atlassian.net/browse/GEOT-7102) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

got wms working

added in query and nativeOnly parameters

fixed test issues

fixed featuresource versus featurecollection issue

cleanup

Got unit tests working

Fixed other tests to use correct hint key

cleanup

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/GeoHashGrid.java [hunk 1]
```diff
@@ -154,26 +163,44 @@ public GridCoverage2D toGridCoverage2D() {
...
-                        LOGGER.fine("Failed to parse aggregation value: " + e);
+        FeatureCollection featureCollection = features;
+        if (features instanceof ForceCoordinateSystemFeatureResults) {
+            ForceCoordinateSystemFeatureResults forceCoordinateSystemFeatureResults =
+                    (ForceCoordinateSystemFeatureResults) features;
+            featureCollection = forceCoordinateSystemFeatureResults.getOrigin();
+        }
+        ElasticBucketVisitor visitor =
+                new ElasticBucketVisitor(aggregationDefinition, queryDefinition, nativeOnly);
+        if (featureCollection instanceof ContentFeatureCollection) {
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/GeoHashGrid.java:181]: This is to handle test cases

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/GeoHashGrid.java [hunk 2]
```diff
@@ -154,26 +163,44 @@ public GridCoverage2D toGridCoverage2D() {
...
+            featureCollection.accepts(visitor, null);
+            return visitor.getBuckets();
+        } else {
+            ObjectMapper mapper = new ObjectMapper();
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/GeoHashGrid.java:185]: Like the previous comment, this is to handle test cases.

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/FilterToElastic.java [hunk 3]
```diff
@@ -1275,9 +1276,15 @@ private Number safeConvertToNumber(Expression expression, Class<?> target) {
     void addViewParams(Query query) {
         if (query.getHints() != null
                 && query.getHints().get(Hints.VIRTUAL_TABLE_PARAMETERS) != null) {
+            throw new UnsupportedOperationException(
+                    "Viewparams not supported with " + "Elasticsearch queries.");
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/FilterToElastic.java]: fixed

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/ElasticParserUtil.java [hunk 4]
```diff
@@ -46,7 +46,7 @@
  * Utilities for parsing Elasticsearch document source and field content to extract values and
  * create geometries.
  */
-class ElasticParserUtil {
+public class ElasticParserUtil {
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/ElasticParserUtil.java]: fixed

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/FilterToElasticException.java [hunk 5]
```diff
@@ -16,7 +16,7 @@
  */
 package org.geotools.data.elasticsearch;
 
-class FilterToElasticException extends RuntimeException {
+public class FilterToElasticException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/FilterToElasticException.java]: fixed

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/ElasticBucketVisitor.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    public ElasticBucketVisitor(
+            String aggregationDefinition, String queryDefinition, Boolean nativeOnly) {
+        this.buckets = new ArrayList<>();
+        this.aggregationDefinition = aggregationDefinition;
+        this.queryDefinition = queryDefinition;
+        this.nativeOnly = nativeOnly;
+    }
+
+    @Override
+    public void visit(Feature feature) {}
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/ElasticBucketVisitor.java]: fixed

File: modules/unsupported/elasticsearch/src/test/java/org/geotools/process/elasticsearch/AbstractContentTest.java [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.LineString;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.Name;
+import org.opengis.filter.Filter;
+
+public abstract class AbstractContentTest {
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/test/java/org/geotools/process/elasticsearch/AbstractContentTest.java:43]: fixed

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/GeoHashGridProcess.java [hunk 8]
```diff
@@ -71,6 +71,20 @@ public GridCoverage2D execute(
                     SimpleFeatureCollection obsFeatures,
 
             // process parameters
+            @DescribeParameter(
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/process/elasticsearch/GeoHashGridProcess.java]: fixed

File: modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/ElasticFeatureSource.java [hunk 9]
```diff
@@ -148,6 +155,80 @@ protected int getCountInternal(Query query) throws IOException {
         return reader;
     }
 
+    @Override
+    public void accepts(
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/ElasticFeatureSource.java]: fixed

File: modules/unsupported/elasticsearch/src/test/java/org/geotools/process/elasticsearch/AbstractContentTest.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (aaime) [reply modules/unsupported/elasticsearch/src/test/java/org/geotools/process/elasticsearch/AbstractContentTest.java:43]: So is this class needed at all now?

[the remark this answers is not in the document]
Comment (turingtestfail) [reply modules/unsupported/elasticsearch/src/test/java/org/geotools/process/elasticsearch/AbstractContentTest.java]: removed

**Pull request merged**